template.tpl

```diff
@@ -453,173 +453,145 @@ const url = 'https://content.zeotap.com/sdk/zeotap.min.js';
 const makeTableMap = require('makeTableMap');
 const getType = require("getType");
 
-function searchWith(array,searchFn){
-  var k=0;
-  array.forEach((value)=>{
-    if(searchFn(value)){k=1;}
-  });
-  return k==1;
-}
-function isNamePresentIn(array,searchKey){
-  return searchWith(array,(value)=>searchKey == value.name);
+function isNamePresentIn(array, searchKey) {
+  for (let i = 0; i < array.length; i++) {
+    if (array[i].name === searchKey) {
+      return true;
+    }
+  }
+  return false;
 }
 
-function removePIIs(listOfPIIs,eventData){
+function removePIIs(listOfPIIs, eventData) {
   log('removing Piis : ', listOfPIIs, eventData);
   var copy = {};
   var piiKey;
-  for(let i=0;i<listOfPIIs.length;i++){
-    piiKey = listOfPIIs[i].name; 
-    if(!! eventData[piiKey]){
+  for (let i = 0; i < listOfPIIs.length; i++) {
+    piiKey = listOfPIIs[i].name;
+    if (!!eventData[piiKey]) {
       eventData[piiKey] = undefined;
     }
   }
-  for(let key in eventData){
-    if(!!eventData[key]){
+  for (let key in eventData) {
+    if (!!eventData[key]) {
       copy[key] = eventData[key];
     }
   }
   log('returning after removing Piis : ', copy);
   return copy;
 }
 
-function mergePreviousData(dataLayer) {
-    var mergedData = dataLayer[dataLayer.length-1];   
-    // Very simple 'merge' that will not overwrite latest
-    for (let d = dataLayer.length-2; d >= 0; d--) {
-      let current = dataLayer[d];
-      if (current.event === undefined) {
-        for (let i in current){
-          if (mergedData[i] === undefined) {
-            mergedData[i] = current[i];
-          }
-        }
-      } else {
-        break; 
-      }
-    }
-    return mergedData;
-  }
-function isNullOrEmpty(o){
-    return o===null || o===undefined;
-} 
-function isEmptyString(s){
-    return isNullOrEmpty(s) || (typeof s === 'string' && s.length<=0);
+function isNullOrEmpty(o) {
+  return o === null || o === undefined;
```

Comment:
can this be simplified with o == undefined?

Reply:
removed fn, since not used